File: petclinic-backend-ts/src/pet-types/dto/pet-type.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsDefined, Length, Min } from 'class-validator';

/**
 * Ported from victor.training.petclinic.rest.dto.PetTypeDto.
 */
export class PetTypeDto {
  @IsDefined({ message: 'must not be null' })
  @Length(1, 80, { message: 'size must be between 1 and 80' })
  @ApiProperty({ example: 'cat', description: 'The name of the pet type.' })
  name!: string;

  // Java @NotNull @Min(0): id is REQUIRED (not read-only). Null -> "must not be null",
  // negative -> "must be greater than or equal to 0".
  @IsDefined({ message: 'must not be null' })
  @Min(0, { message: 'must be greater than or equal to 0' })
  @ApiProperty({ example: 1, description: 'The ID of the pet type.' })
  id!: number;
}

File: petclinic-backend/src/main/java/victor/training/petclinic/rest/dto/PetTypeDto.java
package victor.training.petclinic.rest.dto;

import jakarta.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;

public class PetTypeDto {

    @NotNull
    @Size(min = 1, max = 80)
    @Schema(example = "cat", description = "The name of the pet type.")
    private String name;

    @NotNull
    @Min(0)
    @Schema(example = "1", description = "The ID of the pet type.")
    private Integer id;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }
}
